posthog/middleware.py

```diff
@@ -155,8 +156,26 @@ def __call__(self, request: HttpRequest):
         if request.user.is_authenticated:
             path_parts = request.path.strip("/").split("/")
             project_id_in_url = None
+            current_team = request.user.team
+
+            if len(path_parts) >= 2 and path_parts[0] == "project" and path_parts[1].startswith("phc_"):
+                try:
+                    new_team = Team.objects.get(api_token=path_parts[1])
+
+                    if not self.can_switch_to_team(new_team, request):
+                        raise Team.DoesNotExist
+
+                    path_parts[1] = str(new_team.pk)
+                    return redirect("/" + "/".join(path_parts))
+
+                except Team.DoesNotExist:
+                    if current_team:
+                        path_parts[1] = str(current_team.pk)
```

Comment:
feels better to 404 a link that we can't process than to send to a different team

but it's a valid team they do have access to so 🤷